src/openMVG/multiview/solver_resection_p5pfr_test.cpp

```diff
@@ -0,0 +1,630 @@
+// Copyright (c) 2010 libmv authors.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+
+
+// Copyright (c) 2012, 2013 Pierre MOULON.
+
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <vector>
+#include "openMVG/multiview/solver_resection_p5pfr.hpp"
+#include "testing/testing.h"
+
+using namespace std;
+using namespace openMVG;
+
+bool checkNear(Mat &expect, Mat &actual, double eps)
+{
+  if(expect.rows() != actual.rows() || expect.cols() != actual.cols())
+    return false;
+  for(int i = 0; i < expect.rows(); ++i)
```

Comment:
after the rows check you can do this
return ( (expect - actual).array() .abs() > (eps * expect.abs().array().abs) ).any()
and so much for readability :-)
Otherwise make sure you are using std::abs and for(Mat::Index i etc

The same you can do for the next 2, and better, i think you can just use the one with Vec, as Vec3 is just a fix sized vector